control/server.py

```diff
@@ -127,10 +133,37 @@ def serve(self):
                 gateway_addr, gateway_port))
 
         # Check for existing NVMeoF target state
-        self._restore_state()
+        callbacks = {
+            self.gateway_state.BDEV_PREFIX: {
+                True: gateway_rpc.create_bdev,
+                False: gateway_rpc.delete_bdev
+            },
+            self.gateway_state.SUBSYSTEM_PREFIX: {
+                True: gateway_rpc.create_subsystem,
+                False: gateway_rpc.delete_subsystem
+            },
+            self.gateway_state.NAMESPACE_PREFIX: {
+                True: gateway_rpc.add_namespace,
+                False: gateway_rpc.remove_namespace
+            },
+            self.gateway_state.HOST_PREFIX: {
+                True: gateway_rpc.add_host,
+                False: gateway_rpc.remove_host
+            },
+            self.gateway_state.LISTENER_PREFIX: {
+                True: gateway_rpc.create_listener,
+                False: gateway_rpc.delete_listener
+            }
+        }
+        self.gateway_state.set_update_callbacks(callbacks)
```

Comment:
I don't like that the `gateway_state` has to know about details of a RPC request. I think these callbacks should be more generic and they should contain all the logic of generating a RPC request. So there is no dependencies to protobufs etc. in `gateway_state`.